### PR TITLE
Add action Buttons

### PR DESCRIPTION
**Pull request by jyash97**
- [X] Delete Button

- [x] Clone Button

- [x] Read Button

- [x] Write Button

### CODE REVIEW

File: src/pages/HomePage/ActionButtons.js [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+		);
+	}
+}
+
+const mapStateToProps = state => ({
+	permissions: get(state, '$getAppPermissions'),
+});
+
+const mapDispatchToProps = dispatch => ({
+	fetchPermissions: () => dispatch(getUserAppsPermissions),
```
Comment (bietkul) [inline src/pages/HomePage/ActionButtons.js]: change it to `fetchPermissions: () => dispatch(getUserAppsPermissions())`

File: src/reducers/appsMetricsReducer.js [hunk 2]
```diff
@@ -31,6 +31,15 @@ export default function appsMetricsReducer(
 				error: action.error,
 			};
 		}
+		case APPS.REMOVE_APP_METRICS: {
```
Comment (metagrover) [inline src/reducers/appsMetricsReducer.js]: Why have you created multiple `REMOVE` actions? Ideally, we should create one `REMOVE` action on `APPS` and all the necessary reducers should listen to it, and act accordingly.

File: src/sagas/deleteAppSaga.js [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+}
+
+function* watchDeleteApp(action) {
+	if (action.type === DELETE_APP.LOAD) {
+		yield call(deleteAppWorker, action.payload);
+	}
+}
+
+export default function* deleteAppSaga() {
+	yield takeEvery(DELETE_APP.LOAD, watchDeleteApp);
```
Comment (metagrover) [inline src/sagas/deleteAppSaga.js]: This is weirdly structured imo. We have added unncessary complexity here. Ideally should be done with *one* `DELETE` action that takes the `appId` and `appName` to perform deletion everywhere.

File: src/pages/HomePage/ActionButtons.js [hunk 4]
```diff
@@ -0,0 +1,149 @@
...
+			importFrom: {
+				appname: appName,
+				hosturl: `https://${username}:${password}@${SCALR_URL}`,
+			},
+		};
+		window.open(`${IMPORTER_LINK}${JSON.stringify(parameters)}`, '_blank');
+		e.stopPropagation();
+	};
+
+	copyWriteKey = () => {
```
Comment (metagrover) [inline src/pages/HomePage/ActionButtons.js]: Why do we not have `e.stopPropagation()`s in other action-btns events handlers? 🤔

File: src/constants/index.js [hunk 5]
```diff
@@ -9,10 +9,12 @@ const APPS = {
 	LOAD_METRICS_SUCCESS: 'APPS_LOAD_METRICS_SUCCESS',
 	LOAD_METRICS_FAIL: 'APPS_LOAD_METRICS_FAIL',
 	APPEND: 'APPS_APPEND',
-
 	LOAD_OWNERS: 'APPS_LOAD_OWNERS',
 	LOAD_OWNERS_SUCCESS: 'APPS_LOAD_OWNERS_SUCCESS',
 	LOAD_OWNERS_FAIL: 'APPS_LOAD_OWNERS_FAIL',
+	REMOVE_APP: 'REMOVE_APP',
```
Comment (metagrover) [inline src/constants/index.js]: These should be replaced with one `DELETE` action.

**Comment by siddharthlatest** [comment]
@metagrover Good to go from my end. Thanks @jyash97!

**Comment by siddharthlatest** [comment]
@metagrover I am merging this so I can test the dashboard end-to-end.

**Pull request merged**